### PR TITLE
On MacOS, try to figure out JAVA_HOME if the env var is not set

### PR DESCRIPTION
**Pull request by gongsu832**
Signed-off-by: Gong Su <gong_su@hotmail.com>

### CODE REVIEW

**Comment by gongsu832** [comment]
@jenkins-droid test this please

File: CMakeLists.txt [hunk 1]
```diff
@@ -88,13 +88,28 @@ endif()
...
+    execute_process(
+      COMMAND bash -c "java -XshowSettings:properties -version 2>&1 | \
+                       sed -n 's/[ \t]*java.home = //p'"
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      OUTPUT_VARIABLE JAVA_HOME)
+  endif()
+  if (NOT "${JAVA_HOME}" STREQUAL "")
+    SET(JAVA_INCLUDE_PATH  "${JAVA_HOME}/include")
+    SET(JAVA_INCLUDE_PATH2 "${JAVA_HOME}/include/darwin")
+  endif()
```
Comment (AlexandreEichenberger) [inline CMakeLists.txt:107]: Should we issue a warning when its not found?
Also, same question as asked on another MacOS issue: is the setting of JAVA_HOME documented?

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (gongsu832) [reply CMakeLists.txt:107]: find_package will do that.

**Pull request merged**